contracts/ClaimSYNR.sol

```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract ClaimSYNR is Ownable {
+    mapping(uint32 => bool) public claimed;
+    ERC721 synPass;
+    ERC20 synr;
+    bool public enabled;
+    uint256 public award = 15000 * 10**18;
+
+    constructor(address _synPass, address _synr) {
+        synPass = ERC721(_synPass);
+        synr = ERC20(_synr);
+    }
+
+    // Needs to transfer enough fund to contract before enabling.
+    function enable() external onlyOwner {
+        require(synr.balanceOf(address(this)) == award * 888, "Not enough SYNR");
+        enabled = true;
```

Comment:
If it receives ERC20, the contract should implements `ERC20Receiver`.

Reply:
If not CertiK won't like it :-)

Reply:
it will only receive SYNR one time using unsafe transfer.  it will fail for all safe transfers, which is expected.